Library/Homebrew/api.rb

```diff
@@ -42,19 +42,31 @@ def fetch_json_api_file(endpoint, target:)
       retry_count = 0
       url = "#{Homebrew::EnvConfig.api_domain}/#{endpoint}"
       default_url = "#{HOMEBREW_API_DEFAULT_DOMAIN}/#{endpoint}"
+
+      # TODO: consider using more of Utils::Curl
       curl_args = %W[
         --compressed
         --speed-limit #{ENV.fetch("HOMEBREW_CURL_SPEED_LIMIT")}
         --speed-time #{ENV.fetch("HOMEBREW_CURL_SPEED_TIME")}
       ]
-      curl_args.prepend("--silent") unless Context.current.debug?
+      curl_args << "--progress-bar" unless Context.current.verbose?
+      curl_args << "--verbose" if Homebrew::EnvConfig.curl_verbose?
+      curl_args << "--silent" unless $stdout.tty?
+
+      skip_download = target.exist? &&
+                      !target.empty? &&
+                      (Homebrew::EnvConfig.no_auto_update? ||
+                      ((Time.now - Homebrew::EnvConfig.api_auto_update_secs.to_i) < target.mtime))
 
       begin
         begin
           args = curl_args.dup
           args.prepend("--time-cond", target) if target.exist? && !target.empty?
-          # Disable retries here, we handle them ourselves below.
-          Utils::Curl.curl_download(*args, url, to: target, retries: 0, show_error: false)
+          unless skip_download
+            ohai "Downloading #{url}" if $stdout.tty?
```

Comment:
```suggestion
            ohai "Fetching API data from #{url}" if $stdout.tty?
```
Maybe it's useful to be a bit clearer about what this is?

Reply:
Longer strip plus `ohai` means it's more likely to be truncated. I think the URL itself communicates what it is.